lib/galaxy/model/tool_shed_install/__init__.py

```diff
@@ -79,12 +79,17 @@ def can_reinstall_or_activate(self):
     def get_sharable_url(self, app):
         return common_util.get_tool_shed_repository_url(app, self.tool_shed, self.owner, self.name)
 
-    def get_shed_config_filename(self):
+    @property
+    def shed_config_filename(self):
         shed_config_filename = None
         if self.metadata:
             shed_config_filename = self.metadata.get('shed_config_filename', shed_config_filename)
         return shed_config_filename
 
+    @shed_config_filename.setter
+    def shed_config_filename(self, value):
+        self.metadata['shed_config_filename'] = value
```

Comment:
Should we make sure that `self.metadata` is not `None` by using `self.metadata = metadata or {}` in the class `__init__()` method (line 45)?

Reply:
I think that should work, I'll give it a try